syntax = "proto3";

package client_server;

service ClientServergRPC {
    rpc ReadBlock (ReadBlockReq) returns (ReadBlockReply) {}
    rpc WriteBlock (WriteBlockReq) returns (WriteBlockReply) {}
}
   
// Read block
message ReadBlockReq {
    int32 offset = 1;
}

message ReadBlockReply {
    bytes buf  = 1;
    int32 err = 2;
}

// Write block
message WriteBlockReq {
    int32  offset = 1;
    bytes  buf = 2;
}

message WriteBlockReply {
    int32 err = 1;
}
